vtkext/private/module/vtkF3DOpenGLGridMapper.cxx

```diff
@@ -52,8 +48,8 @@ void vtkF3DOpenGLGridMapper::ReplaceShaderValues(
   );
   vtkShaderProgram::Substitute(VSSource, "//VTK::PositionVC::Impl",
     "gridCoord = vertexMC.xy * fadeDist;\n"
-    "gridOffset = originOffset." + axes2d + ";\n"
-    "gl_Position = MCDCMatrix * vec4(vertexMC." + axes3d + " * fadeDist, 1.0);\n"
+    "gridOffset = originOffset.xz;\n"
+    "gl_Position = MCDCMatrix * vec4(vertexMC.xzy * fadeDist, 1.0);\n"
```

Comment:
I'm not entirely understanding the logic here, but swapping Y and Z makes me think a rotation matrix could be missing in the model matrix.

Reply:
Also, `originOffset` is used as a 2D vector with the Y value ignored. Could we change the uniform to a 2D offset? And maybe use it in the fragment shader directly?

Reply:
> I'm not entirely understanding the logic here, but swapping Y and Z makes me think a rotation matrix could be missing in the model matrix.

not sure either I just reused one of the axis-aligned cases that was already there :/



> Also, `originOffset` is used as a 2D vector with the Y value ignored. Could we change the uniform to a 2D offset? And maybe use it in the fragment shader directly?

you're right, I'll drop `originOffset` and make `gridOffset` a uniform instead

Reply:
> not sure either I just reused one of the axis-aligned cases that was already there :/

Ok, it's probably related to the fact `vertexMC` is 2D in the grid plane (XZ). So using `vertexMC.xzy` is equivalent to `vec3(vertexMC.x, 0, vertexMC.y)`.  
Anyway, that's fine.